### PR TITLE
tsweb: clean up pprof handler registration, document why it's there

### PR DESCRIPTION
**Pull request by danderson**
Updates #cleanup

### CODE REVIEW

File: tsweb/debug.go [hunk 1]
```diff
@@ -48,16 +48,11 @@ func Debugger(mux *http.ServeMux) *DebugHandler {
 	}
 	mux.Handle("/debug/", ret)
 
-	// Register this one directly on mux, rather than using
-	// ret.URL/etc, as we don't need another line of output on the
-	// index page. The /pprof/ index already covers it.
-	mux.Handle("/debug/pprof/profile", BrowserHeaderHandler(http.HandlerFunc(pprof.Profile)))
```
Comment (bradfitz) [inline tsweb/debug.go:54]: was that BrowserHeaderHandler important?

**Review by bradfitz — APPROVED** [review]

File: tsweb/debug.go (hunk 1 above)
Comment (danderson) [reply tsweb/debug.go:54]: Yes, but the debughandler's ServerHTTP does it inline, so no need for a dedicated extra wrapping.

**Comment by danderson** [comment]
Strange, tests are failing now. From the pprof package's documentation, pprof.Index _should_ serve the individual profiles as well as the index of available profiles, but doesn't seem to. I'll have to look more tomorrow.

**Comment by danderson** [comment]
Found it. /debug/pprof/profile is special and not like all the other profile types, in that pprof.Index doesn't know about it. So, we _do_ have to register that handler separately from pprof.Index. I've changed the PR to make that registration happen inline with other pprof things, and document why pprof.Index is not sufficient.

**Review by creachadair — APPROVED** [review]

**Pull request merged**